tools/doc/checkLinks.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const fs = require('fs');
+const { Worker, isMainThread, workerData: path } = require('worker_threads');
+
+function* getLinksRecursively(node) {
+  if (
+    (node.type === 'link' && !node.url.startsWith('#')) ||
+    node.type === 'image'
+  ) {
+    yield node;
+  }
+  for (const child of node.children || []) {
+    yield* getLinksRecursively(child);
+  }
+}
+
+if (isMainThread) {
+  const { extname, join, resolve } = require('path');
+  const DIR = resolve(process.argv[2]);
+
+  console.log('Running Markdown link checker...');
+
+  async function* findMarkdownFilesRecursively(dirPath) {
+    const fileNames = await fs.promises.readdir(dirPath);
+
+    for (const fileName of fileNames) {
+      const path = join(dirPath, fileName);
+
+      const stats = await fs.promises.stat(path);
+      if (
+        stats.isDirectory() &&
+        fileName !== 'api' &&
+        fileName !== 'deps' &&
+        fileName !== 'node_modules'
+      ) {
+        yield* findMarkdownFilesRecursively(path);
+      } else if (extname(fileName) === '.md') {
+        yield path;
+      }
+    }
+  }
+
+  function errorHandler(error) {
+    console.error(error);
+    process.exitCode = -1;
```

Comment:
```suggestion
    process.exitCode = 1;
```

(Exit codes are always positive.)